File: abandon-flow-service/src/main/java/io/github/lagom130/lab/client/config/ClientDecodeConfiguration.java
package io.github.lagom130.lab.client.config;

import feign.codec.Decoder;
import feign.codec.ErrorDecoder;
import io.github.lagom130.lab.client.decoder.ClientErrorDecoder;
import io.github.lagom130.lab.client.decoder.ResultDecoder;
import org.springframework.context.annotation.Bean;

public class ClientDecodeConfiguration {
    @Bean
    public Decoder resultDecoder() {
        return new ResultDecoder();
    }

    @Bean
    public ErrorDecoder clientErrorDecoder() {
        return new ClientErrorDecoder();
    }
}

File: zone-study-java/delivery-bird-service/src/main/kotlin/io/github/lagom130/lab/client/config/ClientDecodeConfiguration.kt
package io.github.lagom130.lab.client.config

import feign.codec.Decoder
import feign.codec.ErrorDecoder
import io.github.lagom130.lab.client.decoder.ClientErrorDecoder
import io.github.lagom130.lab.client.decoder.ResultDecoder
import org.springframework.context.annotation.Bean

class ClientDecodeConfiguration {
    @Bean
    fun resultDecoder(): Decoder {
        return ResultDecoder()
    }

    @Bean
    fun clientErrorDecoder(): ErrorDecoder {
        return ClientErrorDecoder()
    }
}
